src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ConstVal.cs

```diff
@@ -199,5 +128,51 @@ private bool IsDefault(object o)
 
             return false;
         }
+
+        public static ConstVal GetDefaultValue(ConstValKind kind)
+        {
+            switch (kind)
+            {
+                case ConstValKind.Int:
+                    return Zero;
+
+                case ConstValKind.Double:
+                    return new ConstVal(0.0);
+
+                case ConstValKind.Long:
+                    return new ConstVal(0L);
+
+                case ConstValKind.Decimal:
+                    return new ConstVal(0M);
+
+                case ConstValKind.Float:
+                    return new ConstVal(0F);
```

Comment:
I'm not sure, and as I'm not sure I'm leaving it for now.
(Another possibility is lazy-instantiated caching).
Hopefully after more of the sort of changes like merging together duplicate method/properties it'll be a bit easier to form an opinion on that.